storages: added PeclMemcachedStorage using pecl-memcached extension

hrach opened the pull request:
PHP 7.0 has only memcached extension available.

hrach commented:
Well, it seems this repo is disconnected form travis. cc @dg

milo commented on src/Caching/Storages/PeclMemcachedStorage.php:
```diff
@@ -0,0 +1,190 @@
+	{
+		if ($this->memcache->addServer($host, $port, 1) === FALSE) {
+			$error = error_get_last();
+			throw new Nette\InvalidStateException("Memcached::addServer(): $error[message].");
```
Isn't it dangerous? It can kill app when unfortunately fail.

hrach replied:
Well, 
- app without cache actually can kill the server, not only one request
- this is copy & past & modify from the current MemcachedStorage.

hrach commented:
tests are green. the question is. should we do something with the naming. (eg. renamed MemcachedStorage to PeclMemcacheStorage (not the missing D) and create deprecated class that inherits it)

dg commented:
Both are PECL and both are for Memcached, how they differs?

hrach commented:
Memcached is newer, maintained and updated for PHP7, also has richer API.

---

Ad naming:
You have once mentined that the class name is not a name of the extension but the storage name. But in this case there is not sultion to solve it, so I suggest:

Renamed the current MemcachedStorage to PeclMemcacheStorage (note the missing D) and create deprecated class that inherits it with the old one name.

So this storage will be an exception, where the storage implementation names are based on the extension name

JanJakes commented:
Memcache extension is not packaged for PHP7 neither by https://www.dotdeb.org/ nor by `ppa:ondrej/php`. Therefore current MemcachedStorage implementation in Nette can be considered incompatible with PHP7.

dg commented:
Pecl is really not nice, because everything is pecl…

hrach commented:
We be make a BC break, rename the old one to the MemcacheStorage, and in the new one trhow better exception if memcache exception is loaded and not the memcached.

dg commented:
But there is incompatibility… http://php.net/manual/en/book.memcache.php#115666

hrach commented:
I wrote it's a BC break. Nobody claims these storages are interchangeable. Even there is BC break in memcached 5.6 & 7.0 - they have changed the default serializer.

hrach commented:
For the naming, we can be inspired by Doctrine: https://github.com/doctrine/cache/tree/e0ef9e9833ca024d251e77e8d82a00ca92b0aa71/lib/Doctrine/Common/Cache

hrach commented:
@dg, please, let's collaborate your real objections and discuss them now. I have no intention to let this open for months or years, in that case I would do an extension instead. But I strongly believe, this should be in the core/package.

milo commented:
BC break can be solved by prefixing key. What about to hide `Memcache` and `Memcached` into one storage?

hrach commented:
@milo I don't think it's a good idea. It's pretty common to reuse the connection returned by getConnection.

dg commented:
To rename current MemcachedStorage is imho not needed, it will be removed in next Nette (for PHP 7), or it can be replaced with this storage. Currently we can use name as NewMemcachedStorage or something similar. It is like mysql & mysqli.

hrach commented:
If it's temporary, I like PeclMemcached more. But no objections to NewMemcachedStorage. Better something than nothing. So, pick a name, I will rename it and rebase it :)

dg commented:
Yes, naming things is hard.

I don't like `Pecl` because it says nothing about how they differ. Moreover they both are PECL. 

New extension uses libMemcached, so what about LibMemcached?

hrach commented:
Thanks. :+1:
